src/store/fences/selector.ts

```diff
@@ -6,3 +9,24 @@ export type FencesProps = initialState;
 
 export const fencesSelector = (state: RootState) => state.fences;
 export const fencesAuthorizedStudiesSelector = (state: RootState) => state.fences.authorizedStudies;
+export const fencesAtLeastOneAuthentificationConnectedSelector = (state: RootState) =>
+  Object.keys(FENCE_NAMES).some((fenceKey) => {
+    const key = fenceKey as FENCE_NAMES;
+    if (state.fences[key] === undefined) {
+      return false;
+    }
```

Comment:
I think you could but add this condition in your return:
`return state.fences[key] && state.fences[key].status === FENCE_AUTHENTIFICATION_STATUS.connected;`

Reply:
je préfère utiliser les earlier return, je trouve la lecture plus simple